poppy/wfe.py

```diff
@@ -579,7 +579,249 @@ def get_opd(self, wave):
         
         self.opd = opd
         return self.opd
+
+
+class KolmogorovWFE(WavefrontError):
+    """ A turbulent phase screen.
+    
+    This is an implementation of a turbulent phase screen as by the
+    Kolmogorov theory of turbulence.
+    
+    Parameters
+    -----------------
+    r0 : astropy.quantity
+        Fried parameter (m).
+    
+    Cn2 : astropy.quantity
+        Index-of-refraction structure constant (m^{-2/3}).
+    
+    dz : astropy.quantity
+        Propagation distance (m).
+    
+    inner_scale : astropy.quantity
+        Inner scale of the turbulence (m). The inner scale affects the
+        calculation results if kind = 'von Karman', 'Tatarski', or 'Hill'.
+    
+    outer_scale : astropy.quantity
+        Outer scale of the turbulence (m). The outer scale only affects the
+        calculation results if kind='von Karman'.
+    
+    kind : string
+        Kind of the spatial power spectrum. Must be one of 'Kolmogorov',
+        'Tatarski', 'von Karman', 'Hill'.
     
+    References
+    -------------------
+    For a general overview of the Kolmogorov theory, read
+    L. C. Andrews and R. L. Phillips, Laser Beam Propagation Through Random
+    Media, 2nd ed. (Society of Photo Optical, 2005).
+    
+    Other relevant references are mentioned in the respective functions.
+    """
+    
+    @utils.quantity_input(r0=u.meter, Cn2=u.meter**(-2/3), dz=u.meter,
+                          inner_scale=u.meter, outer_scale=u.meter)
+    def __init__(self, name="Kolmogorov WFE", r0=None, Cn2=None, dz=None,
+                 inner_scale=None, outer_scale=None, kind='Kolmogorov', **kwargs):
+        
+        if dz is None and not all(item is not None for item in [r0, Cn2]):
+            raise ValueError('To prepare a turbulent phase screen, \
+                             dz and either Cn2 or r0 must be given.')
+        
+        super(KolmogorovWFE, self).__init__(name=name, **kwargs)
+        
+        self.r0 = r0
+        self.Cn2 = Cn2
+        self.dz = dz.to(u.m)
+        self.inner_scale = inner_scale
+        self.outer_scale = outer_scale
+        self.kind = kind
+    
+    def get_opd(self, wave):
```

Comment:
Thanks for including that nice set of tests for the Kologorov code. One piece seems to have slipped through the cracks: actually calling the `get_opd` function as part of any of those tests. The Codecov test coverage report (see comments added onto this PR automatically by that) shows that whole function is untested. It would be good to include a test of this, even if a very basic one such as just demonstrating that adding a Kolmogorov phase screen reduces the Strehl of a beam by at least some amount.  

Other than that the test coverage looks excellent. The existing tests cover all the sub components where most of the complexity is; but we should ideally also have a test of the overall top level function to show that puts those components together correctly. Or at least in a way that lets that function be called and return reasonable results by some simple check. Thank you!

Reply:
Hi @mperrin. I just updated my code to include a test of get_opd. However, the checks that run were not all successful. As in my other commit, one test complains about a decrease in coverage of files I never touched! I am not that experienced with version control etc, please advice what I can do to solve that issue.

Reply:
Hi! First, we have seen that bug before from the code coverage system, where it complains about decreased coverage in files never touched. Not sure what's going on with that, but it's definitely a bug, and we generally tend to ignore it when that happens. 

Secondly, on my side right now I don't even see that - it shows all tests passing, including the code coverage over the diff. Transient glitch apparently? While that's not entirely reassuring, it does tend to reinforce our habit of just ignoring that bug when it crops up!